### PR TITLE
Implement named VIP support, with some refactoring to make it fit

### PR DESCRIPTION
**Pull request by nickbp**
- ConnectionResource: Add 'vip' entry mirroring Kafka's, plus better tests
- CassandraDaemonTask: Add a dedicated Factory which handles whether named VIPs are available

### CODE REVIEW

File: cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/resources/TasksResource.java [hunk 1]
```diff
@@ -141,8 +145,9 @@ public void replace(@QueryParam("node") final String name)
     @GET
     @Path("connect")
```
Comment (gabrielhartmann) [inline cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/resources/TasksResource.java]: I thought this info should be at the `/connection` endpoint.

**Comment by gabrielhartmann** [comment]
LGTM if tests pass.

**Comment by mrbrowning** [comment]
Also seems fine to me. In the future, can you split out preparatory refactoring into its own PR? That would lower the likelihood of review fatigue leading to missed defects.

File: cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/resources/TasksResource.java (hunk 1 above)
Comment (nickbp) [inline cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/resources/TasksResource.java]: I think this interface is left here for compatibility reasons, the calls were all marked @Deprecated and just call into ConnectionResource, which has the `/connection` endpoint.

Comment (gabrielhartmann) [inline cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/resources/TasksResource.java]: Ok, there's an API audit epic here: https://mesosphere.atlassian.net/browse/INFINITY-514.  We'll catch inconsistencies there.

**Comment by nickbp** [comment]
Current test failures are genuine - need to update tests to reflect the new vip entry in /connection

**Comment by nickbp** [comment]
Build is now green. Merging at 5pm unless there's any objections.

File: cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/resources/TasksResource.java (hunk 1 above)
Comment (mohitsoni) [inline cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/resources/TasksResource.java]: Please see https://github.com/mesosphere/dcos-cassandra-service/pull/233 for deprecation notes.

File: cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/tasks/CassandraDaemonTask.java [hunk 2]
```diff
@@ -248,5 +293,26 @@ public CassandraDaemonTask updateId() {
...
+            if (!capabilities.supportsNamedVips()) {
+                return null;
+            }
+        } catch (Exception e) {
+            LOGGER.warn("Unable to determine Named VIP support, assuming they are unavailable.", e);
+            return null;
+        }
+        DiscoveryInfo.Builder discoveryBuilder = DiscoveryInfo.newBuilder()
+            .setVisibility(DiscoveryInfo.Visibility.EXTERNAL)
+            .setName(nodeName);
```
Comment (mohitsoni) [inline cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/tasks/CassandraDaemonTask.java]: I think we should namespace the name attribute. For ex: Running multiple Cassandra clusters in the same Mesos cluster.
Ex: https://github.com/verma7/dcos-cassandra-service/commit/f17c31a89eb978afc47d83ed6752fc2fb04c51c9#diff-da60efe1f8e7ad2a82d142a2d97f3d5eR630

**Pull request merged**